### PR TITLE
Emit retryable in the JSON error envelope

### PR DESCRIPTION
**Pull request by jeremy**
## What

Add a top-level boolean `retryable` to the `--json` error envelope, mirroring the error's Retryable classification: always present on error envelopes, never on success ones.

## Why

The SDK and the CLI both classify failures as retryable, and `output.Error` carries that flag (`internal/output/errors.go:49` copies `sdkErr.Retryable`), but the `--json` error envelope dropped it. A consumer of the envelope could not tell a transient failure from a verdict without keeping its own list of codes and message spellings — which is exactly what `basecamp-local-agent-connector` had to do (`TRANSIENT_CODES` + a message regex), and which drifts the moment either side changes wording.

## Contract

The error envelope carries a top-level boolean `retryable`:

```json
{"ok": false, "error": "Gateway error (503)", "code": "api_error", "retryable": true, "hint": "..."}
```

Contract:
- **Always present on an error envelope**, `false` when nothing classified the error as retryable (no positive signal ⇒ not retryable).
- **Never present on a success envelope.**
- Field order: `ok, error, code, retryable, hint, meta` — existing fields keep their positions; the field is additive.
- Applies to every sink that emits the error object: `--json`, `--agent`/`--quiet` (`{ok:false,...}`), and `--jq` (filter runs on the envelope). Styled/markdown renderers are unchanged.

`basecamp-local-agent-connector` will key on this field when present and fall back to its list only for older CLIs that omit it.

## Where `Retryable` is set (audit)

`true`:
- SDK `checkResponse` (generated service layer, `helpers.go`): 429; every 5xx except 507.
- SDK `singleRequest` (raw client): 429 (`ErrRateLimit`); 502/503/504 gateway errors; `http.Client.Do` failure → `ErrNetwork` (covers DNS, connection refused, TLS, and client/context timeouts).
- SDK `ErrRateLimit`, `ErrNetwork` constructors; `oauth/discovery.go` and `oauth/device_errors.go` network failures.
- SDK internal "Token refreshed" marker after a 401 refresh (consumed by the retry loop).
- Shared `basecamp/cli` `output.ErrRateLimit`, `output.ErrNetwork`.
- CLI `convertSDKError` (`internal/commands/projects.go`, `internal/names/resolver.go`): `ErrRateLimited`, `ErrCircuitOpen`, `ErrBulkheadFull`; SDK errors pass their flag through.
- Batch commands (`comment.go`, `todos.go`, `todolists.go`, `assign.go`) preserve `outErr.Retryable` when re-wrapping the first API error.

`false` (explicit or by default): usage, not found, auth, forbidden, validation (400/422), 507 account limit, raw-client 500 (`ErrAPI(500, …)` — the raw client deliberately does not retry 500; the generated service layer does mark 500 retryable, so `retryable` on a 500 depends on which path the command took, mirroring the SDK's own retry behaviour), and anything unclassified (`AsError` fallback).

## Testing

- [x] `make check` passes (run as `bin/ci` on a 32-core builder; see below)

- `internal/output`: retryable SDK error → `true`; `ErrNotFound` → key present and `false`; CLI `ErrRateLimit`/`ErrNetwork`/wrapped → `true`; unclassified error → `false`; field-order pin; quiet-mode envelope carries it; success envelope omits it.
- `internal/commands`: `projects create` against an httptest server answering 503 → `retryable: true`, 404 → `false` (SDK → `convertSDKError` → `app.Err`). A mutation is used because the generated client retries idempotent reads on its own 1s/2s schedule, which is not under test here.
- `e2e/errors.bats`: the usage-envelope test asserts `.retryable == false`; a new test runs the real binary against a python stub answering 503 and asserts `.retryable == true`, `.code == api_error`.

All of these fail on `main` (checked by stashing `envelope.go`): the Go tests fail on the missing key, the bats tests see `null`.

## Declined alternatives

- **`omitempty` on `retryable`** — would make `false` indistinguishable from "old CLI, field unknown", which is the exact ambiguity the connector's fallback exists to resolve. Always-present on errors is the contract.
- **Put it under `meta`** — `meta` is for request-scoped diagnostics (`request_id`, `stats`); whether to retry is a property of the error itself and belongs beside `code`.
- **Emit `retryable: false` on success envelopes** for symmetry — meaningless on success and would change every success payload; consumers gate on `ok` first.
- **Also emit a `retry_after`** — the SDK does not surface the parsed `Retry-After` on the error (it consumes it in its own backoff), so there is nothing truthful to put there yet; a follow-up if the SDK exposes it.
- **Reconcile the raw-client 500 vs service-layer 500 classification** — that is SDK behaviour, and the envelope should report the SDK's verdict rather than second-guess it.

- **Preserve envelope field order on TTY JSON output** (Codex P2) — the TTY path re-encodes through map-backed sanitization (#479) and alphabetizes keys for *every* envelope field, predating `retryable`. Machine consumers (the connector, scripts, `--jq`) read piped output, which keeps struct order; the TTY rendering serves a human whose parser is their eyes, and JSON consumers must not key on order anyway. The order-pin test comment now scopes the contract to the piped representation instead. Also declined pinning the alphabetical TTY order in a test — that would enshrine incidental behavior as contract.

### CODE REVIEW

**Comment by jeremy** [comment]
@codex review

File: internal/output/envelope.go [hunk 1]
```diff
@@ -55,12 +55,21 @@ type Breadcrumb struct {
 }
 
 // ErrorResponse is the error envelope for JSON output.
+//
+// Retryable is always present on an error envelope and never on a success
+// one: true when the failure was classified transient (network, timeout,
+// rate limit, 5xx/gateway, circuit open, bulkhead full), false for a verdict
+// (usage, not found, auth, forbidden, validation, account limit) and for any
+// error nothing classified. Consumers key on the field rather than on the
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/output/envelope.go:64]: Fixed in 08b86a50: the doc comment now reads "most 5xx/gateway responses — not all: 507 and a raw-client 500 are verdicts", stating the classification as a verdict with its exceptions rather than a blanket 5xx rule.

File: README.md [hunk 2]
```diff
@@ -134,6 +134,25 @@ Every command supports `--json` for structured output:
...
+  "retryable": true,
+  "hint": "..."
+}
+```
+
+`retryable` is present on every error envelope — `true` for transient failures
+(network, timeout, rate limit, 5xx/gateway, circuit open), `false` for a verdict
+(usage, not found, auth, forbidden, validation, account limit) — and never on a
+success envelope. Key on it rather than on the code or message when deciding
+whether to retry.
```
[the remark this answers is not in the document]
Comment (jeremy) [reply README.md]: Fixed in 08b86a50: README now says "most 5xx/gateway responses — not all: 507 and some 500s are verdicts" and frames the whole list as the classifier's verdict. Same sweep applied to SKILL.md and the ErrorResponse doc comment.

File: skills/basecamp/SKILL.md [hunk 3]
```diff
@@ -1349,15 +1349,21 @@ the specific argument. Use this for elicitation:
...
+{"ok": false, "error": "<content> required", "code": "usage", "retryable": false, ...}
 ```
 
 The `error` field names the missing `<arg>` — use it to prompt the user for the specific value.
 
+**Retryable errors (`retryable`):** every error envelope carries a boolean `retryable` —
+`true` when the failure is transient (network, timeout, rate limit, 5xx/gateway, circuit
+open) and a retry can change the outcome, `false` for a verdict (usage, not found, auth,
+forbidden, validation, account limit). Key on it rather than on `code` or `error` when
+deciding whether to retry; it is never present on a success envelope.
```
[the remark this answers is not in the document]
Comment (jeremy) [reply skills/basecamp/SKILL.md]: Fixed in 08b86a50: the SKILL.md paragraph now phrases retryable as the CLI's classification verdict ("most 5xx/gateway responses — not all: 507 and some 500s are verdicts") and adds that false also covers unclassified failures — no known reason a retry would help, not proof of permanence.

File: internal/output/envelope.go [hunk 4]
```diff
@@ -55,12 +55,21 @@ type Breadcrumb struct {
...
 type ErrorResponse struct {
-	OK    bool           `json:"ok"`
-	Error string         `json:"error"`
-	Code  string         `json:"code"`
-	Hint  string         `json:"hint,omitempty"`
-	Meta  map[string]any `json:"meta,omitempty"`
+	OK        bool           `json:"ok"`
+	Error     string         `json:"error"`
+	Code      string         `json:"code"`
+	Retryable bool           `json:"retryable"`
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/output/envelope.go:71]: Took the relax-and-scope option in 08b86a50. The TTY re-encode alphabetizing keys predates this field (#479's sanitization pass) and applies to every envelope field, not just retryable; machine consumers — the connector, scripts, --jq pipelines — read piped output, which is non-TTY and keeps struct order. The order-pin test comment now scopes the contract to the piped representation and documents that the TTY path alphabetizes. Declined building an order-preserving sanitizer for a human-facing cosmetic path, and declined pinning the alphabetical TTY order in a test — that would enshrine incidental behavior as contract.

**Comment by jeremy** [comment]
Since last push (`08b86a50`), converging the review round:

**Applied**
- **Docs: unclassified fallback** — README, SKILL.md, and the `ErrorResponse` doc comment now state that `retryable: false` also covers failures nothing classified: it means "no known reason a retry would help", not proof of permanence.
- **Docs: 5xx overstatement** (all three Copilot threads, swept as one class) — the blanket "5xx/gateway ⇒ retryable" phrasing is now "most 5xx/gateway responses — not all: 507 and a raw-client 500 are verdicts", in the same three sites.
- **Field-order pin scoped** (Codex P2) — took the offered relax-and-scope option: the test comment now pins the piped (non-TTY) representation machine consumers read, and documents that the TTY path's map-backed sanitization re-encode alphabetizes keys (pre-existing since #479).
- **e2e stub leak** — `start_unavailable_api_stub`'s startup-timeout branch now stops the stub before returning, and the python process runs with stdout/stderr to a log file (dumped on failure, matching the recorder helper) and FD 3 closed, so a wedged stub can no longer orphan a listener or hold the bats run open. Verified both directions with a forced-timeout harness: the pre-fix helper leaves the stub running past `return 1`; the fixed helper kills it.

**Declined**
- Building an order-preserving sanitizer for the TTY JSON path, or test-pinning its alphabetical order — recorded in the PR body's Declined section.

**Tests**: `internal/output` and full `e2e/errors.bats` (48/48) green; `make check` Go legs green except the pre-existing PTY-dependent interactivity tests, which fail identically on the previous head in this sandboxed environment (no real TTY) and are untouched by this PR.

**Comment by jeremy** [comment]
@codex review

File: internal/output/output_test.go [hunk 5]
```diff
@@ -470,6 +470,125 @@ func TestWriterErrIncludesRequestIDMeta(t *testing.T) {
...
+	assert.Equal(t, CodeAPI, decoded["code"])
+	assert.Equal(t, false, decoded["retryable"], "no positive signal means not retryable")
+}
+
+// Field order is part of the piped-envelope contract: retryable sits between
+// code and hint, and the existing fields keep their positions. The pin covers
+// the non-TTY representation machine consumers read; on a TTY, writeJSON
+// re-encodes the envelope through map-backed sanitization, so keys come out
+// alphabetized there — that path serves a human reader and carries no order
+// contract.
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/output/output_test.go:541]: Fixed in e66687d0 — by narrowing the contract, the second remedy: README now states key order is not part of the envelope contract (the TTY path re-encodes through a map and alphabetizes; that predates this PR — #479) and consumers match on key names. Preserving order through the TTY sanitizer was declined in the body's Declined section: piped/machine output preserves struct order, and pinning the TTY's alphabetical order would enshrine incidental behavior.
